gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -118,7 +117,19 @@ public Blob(Storage storage, BlobInfo info) {
    */
   public Blob(Storage storage, String bucket, String blob) {
     this.storage = checkNotNull(storage);
-    this.info = BlobInfo.of(checkNotNull(bucket), checkNotNull(blob));
+    this.info = BlobInfo.builder(BlobId.of(bucket, blob)).build();
```

Comment:
I wonder (though not sure about it) if in the cases where `BlobInfo` is not provided in the constructor (here and bellow) we should actually get it from the service, and use empty if not exists (and document that). 

If you don't think we should do that then we should probably make that clear in the documentation (that in these cases `BlobInfo` is basically empty).

Also, unrelated to this PR but I just noticed that. I think we should add `BlobSourceOption...` for `exists`.